packages/vitest/src/node/workspace.ts

```diff
@@ -319,12 +319,28 @@ export class WorkspaceProject {
 
   getSerializableConfig() {
     const optimizer = this.config.deps?.optimizer
+    const poolOptions = this.config.poolOptions
+
+    // Use isolate from root only if it's explicitly disabled. Otherwise fallback to other options.
+    const isolate = this.config.isolate === false ? false : undefined
```

Comment:
Maybe we can resolve this in `setServer` before we resolve the config?

Reply:
Default value is set already before `setServer`:

https://github.com/vitest-dev/vitest/blob/a73c1c2ca3a3ca93b317657b9328889434c36344/packages/vitest/src/node/plugins/workspace.ts#L101-L106

But what we could do is resolve `isolate` from `this.server` instead here:

```suggestion
    // Resolve from server.config to avoid comparing against default value
    const isolate = this.server?.config?.test?.isolate
```

Reply:
Sounds good to me